pyop2/mpi.py

```
@@ -232,18 +470,8 @@ def free_comms():
 
 def hash_comm(comm):
     """Return a hashable identifier for a communicator."""
-    # dup_comm returns a persistent internal communicator so we can
-    # use its id() as the hash since this is stable between invocations.
-    return id(dup_comm(comm))
-
-
-def collective(fn):
-    extra = trim("""
-    This function is logically collective over MPI ranks, it is an
-    error to call it on fewer than all the ranks in MPI communicator.
-    """)
-    fn.__doc__ = "%s\n\n%s" % (trim(fn.__doc__), extra) if fn.__doc__ else extra
-    return fn
+    assert is_pyop2_comm(comm)
```

Again, probably better to raise an exception.